controller.py

```diff
@@ -81,46 +83,157 @@
 
 OPDS_2_MEDIA_TYPE = 'application/opds+json'
 
+class MetadataWrangler(object):
+    """The metadata wrangler itself.
 
-def authenticated_client_from_request(_db, required=True):
-    header = request.headers.get('Authorization')
-    if header and 'bearer' in header.lower():
-        shared_secret = base64.b64decode(header.split(' ')[1])
-        client = IntegrationClient.authenticate(_db, shared_secret)
-        if client:
-            return client
-    if not required and not header:
-        # In the case that authentication is not required
-        # (i.e. URN lookup) return None instead of an error.
-        return None
-    return INVALID_CREDENTIALS
+    A simple grouping of controllers.
+    """
 
+    def __init__(self,_db):
+        self._db = _db
+        self.heartbeat = HeartbeatController()
+        self.canonicalization =  CanonicalizationController(self._db)
+        self.index = IndexController(self._db)
+        self.urn_lookup = URNLookupController(self._db)
+        self.catalog = CatalogController(self._db)
+        self.integration = IntegrationClientController(self._db)
 
-def collection_from_details(_db, client, collection_details):
-    if not (client and isinstance(client, IntegrationClient)):
-        return None
+    @classmethod
+    def authenticated_client_from_request(cls, _db, required=True):
+        """Look up the IntegrationClient that's making the active
+        request.
```

Comment:
This is a class method because that makes it easy to call in different situations (testing vs not, decorator vs inside a controller).